modules/dns_common/dns_common.rb

```diff
@@ -170,10 +170,10 @@ def aaaa_record_conflicts(fqdn, ip)
       end
     end
 
-    def cname_record_conflicts(fqdn, target)
-      current = resolver.getresources(fqdn, Resolv::DNS::Resource::IN::CNAME)
+    def cname_record_conflicts(fqdn, host_alias)
+      current = resolver.getresources(host_alias, Resolv::DNS::Resource::IN::CNAME)
```

Comment:
I agree with @ekohl, mostly. As this is internal api, deprecating a call and introducing a new one *is* an option: this way provider developers have about three months to catch up. Having said this, I'd rate good method parameter names higher than consistent parameter order, esp. when we only have a couple of parameters in these methods.